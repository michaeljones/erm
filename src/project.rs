use std::path::PathBuf;

pub struct Settings {
    pub source_directories: Vec<PathBuf>,
}

impl Settings {
    pub fn new() -> Self {
        Self {
            source_directories: vec![],
        }
    }
}

impl Default for Settings {
    fn default() -> Self {
        Self::new()
    }
}
