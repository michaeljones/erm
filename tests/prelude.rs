mod common;

mod prelude {

    use std::path::PathBuf;

    use erm::project;

    use crate::common::eval;

    #[test]
    fn use_built_in_string_module() {
        // To check that something like the String module can exist in the prelude and be accessed in
        // general functions without importing it
        let src = r#"
        module Main exposing (..)
        main args =
          String.append "Hello, " "World"
        "#;
        let result = eval(src, None);
        insta::assert_snapshot!(result);
    }

    #[test]
    fn imported_module_can_use_prelude() {
        let src = r#"
        module Main exposing (..)
        import Impl.Test
        main args =
          Impl.Test.hello_from_prelude
        "#;

        let settings = project::Settings {
            source_directories: vec![PathBuf::from("tests/modules")],
        };

        let result = eval(src, Some(settings));
        insta::assert_snapshot!(result);
    }
}
