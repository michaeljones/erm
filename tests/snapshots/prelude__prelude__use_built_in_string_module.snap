---
source: tests/prelude.rs
expression: result

---
Hello, World
